src/views/App/BgMusic/BgMusic.tsx

```diff
@@ -0,0 +1,65 @@
+import { preferencesViewModel } from '@/services/preferencesViewModel';
+import playlist from '../musicAssets';
+import { FC } from 'react';
+import { observer } from 'mobx-react';
+
+export let audio: HTMLAudioElement | undefined
+
+export const BgMusic: FC = observer(({})=>{
+    const {soundMode} = preferencesViewModel;
+    let {isPlay} = preferencesViewModel
+    let currentSong = -1
+
+    const playPauseClick = ()=>{
+        currentSong = -1;
+        if (isPlay && audio) {
+            audio.pause();
+            isPlay = !isPlay 
+        } 
+        else if(soundMode){
+            playcurrent();
+            isPlay = !isPlay
+        }
+    }
+
+    const getCurr = ()=>{
+        if (currentSong < playlist.length-1) {
+            currentSong+=1
+        }else{
+            currentSong=0
+        }
+        return currentSong
+    }
+    const playcurrent = (_currentSong = getCurr())=>{
+        audio = new Audio(playlist[_currentSong]);
+        audio.addEventListener("ended", () => {
+            playcurrent
```

Comment:
This won't work. It should be a function call (missed the parentheses), or even better to remove the anonymous function and just write playcurrent:
instead of () => { playcurrent() } just put playcurrent,

Reply:
Hi, thanks for the comment, i updated